dashboard/app/helpers/levels_helper.rb

```diff
@@ -41,8 +41,16 @@ def set_videos_and_blocks_and_callouts_and_instructions
     select_and_track_autoplay_video
 
     if @level.is_a? Blockly
-      @toolbox_blocks = @toolbox_blocks || @level.toolbox_blocks
-      @start_blocks = initial_blocks(current_user, @level) || @start_blocks || @level.start_blocks
+      @toolbox_blocks =
+        @toolbox_blocks ||
+        @level.try(:project_template_level).try(:toolbox_blocks) ||
+        @level.toolbox_blocks
+
+      @start_blocks =
+        initial_blocks(current_user, @level) || # check if this level inherits solution from previous level
+        @start_blocks ||
+        @level.try(:project_template_level).try(:start_blocks) ||
+        @level.start_blocks
```

Comment:
Nit: prefer continuations to have joining token at the beginning for context, e.g.:
```ruby
@start_blocks = initial_blocks(current_user, @level)
  || @start_blocks # check if this level inherits solution from previous level
  || @level.try(:project_template_level).try(:start_blocks)
  || @level.start_blocks
```
(Ditto above)